core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -1341,13 +1335,15 @@ class Partition(val topicPartition: TopicPartition,
     isrState = proposedIsrState
 
     if (!alterIsrManager.submit(alterIsrItem)) {
-      // If the ISR manager did not accept our update, we need to revert back to previous state
+      // If the ISR manager did not accept our update, we need to revert back to previous state.
```

Comment:
"previous state" seems like the wrong description of what we're reverting to here.  After all, we're NOT reverting back to the state that existed before the ISR was updated by whatever change invalidated our AlterIsrRequest.  "current state" seems more accurate.

Also, this comment seems unnecessarily ZK-specific.  Can we mention "ChangePartitionRecord or LeaderAndIsrRequest" as candidates for altering the ISR?

Reply:
How about

> If the ISR manager did not accept our update, we need to revert the proposed state.

I'm not quite sure what you mean by ZK-specific. When I say "ISR state was updated", I really mean "when Partition#isrState was updated". How about:

> This can happen if the Partition's ISR state is changed due to handling a LeaderAndIsrRequest from the controller
> while an AlterIsrRequest is still in-flight. See KAFKA-12686 for details

Reply:
KRaft controllers do not make LeaderAndIsr requests.  So the description here is specific to the ZK-based controller.  I was suggesting that we either make this more generic or talk about both the old and new ways of updating the state.

Reply:
Thanks for the clarification, I see what you meant now. I've updated the comment.